elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala

```diff
@@ -28,12 +32,49 @@ trait MappingHandlers {
       }
     }
 
-
     override def build(request: GetMappingRequest): ElasticRequest = {
       val endpoint = request.indexes match {
         case Indexes(Nil)       => "/_mapping"
         case Indexes(indexes)   => s"/${indexes.mkString(",")}/_mapping"
       }
+
+      ElasticRequest("GET", endpoint)
+    }
+  }
+
+  implicit object GetFieldMappingRequest extends Handler[GetFieldMappingRequest, Seq[IndexFieldMapping]] {
+    override def responseHandler: ResponseHandler[Seq[IndexFieldMapping]] = new ResponseHandler[Seq[IndexFieldMapping]] {
+
+      override def handle(response: HttpResponse) : Either[ElasticError, Seq[IndexFieldMapping]] = response.statusCode match {
+        case 201 | 200       =>
+          val raw = ResponseHandler.fromResponse[Map[String, Map[String, Map[String, Map[String, Any]]]]](response)
+          Right(raw.map {
+            case (index, types) =>
+              val mappings = types("mappings").map {
+                case (_, mapping) => FieldMapping(mapping("full_name").toString, mapping("mapping").asInstanceOf[Map[String, Any]])
```

Comment:
This method is largely a copy pasta, but this part is different as it's deserializing an object that looks like this:

```
{
    "index": {
        "mappings": {
            "a": {
                "full_name": "a",
                "mapping": {
                    "a": {
                        "type": "text",
                        "store": true,
                        "analyzer": "whitespace"
                    }
                }
            }
        }
    }
}
```